src/Cardano/Address/Style/Icarus.hs

```diff
@@ -296,6 +306,33 @@ deriveAddressPublicKey =
 -- >
 -- > bech32 $ paymentAddress icarusMainnet (toXPub <$> addrK)
 -- > "addr1vxpfffuj3zkp5g7ct6h4va89caxx9ayq2gvkyfvww48sdncxsce5t"
+--
+-- | Analyze an 'Address' to know whether it's an Icarus address or not.
+-- Returns 'Nothing' if the address isn't a byron address, or return a string
+-- ready-to-print that gives information about an address.
+--
+-- @since 2.0.0
+inspectIcarusAddress :: Address -> Maybe String
+inspectIcarusAddress addr = do
```

Comment:
I wonder if we cannot reuse some parts of body here, it is very close to what we have in `inspectByronAddress`

Reply:
To some extend, but both are still different and it's not really like they share logic. What applies to doesn't necessarily apply to the other; so I find it easier to keep them fully separated.